File: datasets/IR-Plag/convertai-original-java-rest-kotlin/case-03/plagiarized/L6/04/Kasus3.kt
package IR

import java.util.Scanner

/**
 * @author C166AB8FBBBE55174E16A67530426A8A 19011F43894F06E2BA0CA74061427469
 */
fun hitungTinggi(feet: Double, inches: Double): Double {
    return feet * 12 + inches
}

fun hitungBmi(weight: Double, feet: Double, inches: Double): Double {
    return weight * 0.45359237 / ((hitungTinggi(feet, inches) * 0.0254) * (hitungTinggi(feet, inches) * 0.0254))
}

fun cekBmi(bmi: Double): String {
    return when {
        bmi < 18.5 -> "Underweight"
        bmi < 25 -> "Normal"
        bmi < 30 -> "Overweight"
        else -> "Obese"
    }
}

fun main() {
    val input = Scanner(System.`in`)
    // Prompt the user to enter weight in pounds
    print("Enter weight in pounds: ")
    val weight = input.nextDouble()
    // Prompt the user to enter height 
    print("Enter feet: ")
    val feet = input.nextDouble()
    print("Enter inches: ")
    val inches = input.nextDouble()
    // Display result
    println("BMI is ${hitungBmi(weight, feet, inches)}")
    println(cekBmi(hitungBmi(weight, feet, inches)))
}

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-03/plagiarized/L6/04/Kasus3.ts
import readline from 'readline';

function hitungTinggi(feet: number, inches: number): number {
    return feet * 12 + inches;
}

function hitungBmi(weight: number, feet: number, inches: number): number {
    return weight * 0.45359237 / ((hitungTinggi(feet, inches) * 0.0254) * (hitungTinggi(feet, inches) * 0.0254));
}

function cekBmi(bmi: number): string {
    if (bmi < 18.5) {
        return "Underweight";
    } else if (bmi < 25) {
        return "Normal";
    } else if (bmi < 30) {
        return "Overweight";
    } else {
        return "Obese";
    }
}

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Enter weight in pounds: ", (weightStr) => {
    const weight = parseFloat(weightStr);
    rl.question("Enter feet: ", (feetStr) => {
        const feet = parseFloat(feetStr);
        rl.question("Enter inches: ", (inchesStr) => {
            const inches = parseFloat(inchesStr);
            const bmi = hitungBmi(weight, feet, inches);
            console.log("BMI is " + bmi);
            console.log(cekBmi(bmi));
            rl.close();
        });
    });
});

